ionoscloud/resource_dbaas_mariadb_cluster.go

```diff
@@ -0,0 +1,245 @@
+							Type:             schema.TypeString,
+							Description:      "The numeric LAN ID to connect your cluster to.",
+							Required:         true,
+							ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotWhiteSpace),
+						},
+						"cidr": {
+							Type:             schema.TypeString,
+							Description:      "The IP and subnet for your cluster.",
+							Required:         true,
+							ValidateDiagFunc: validation.ToDiagFunc(VerifyUnavailableIPs),
+						},
+					},
+				},
+			},
+			"display_name": {
+				Type:        schema.TypeString,
+				Description: "The friendly name of your cluster.",
+				Required:    true,
+			},
+			"credentials": {
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Description: "Credentials for the database user to be created.",
+				Required:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"username": {
+							Type:             schema.TypeString,
+							Description:      "The username for the initial MariaDB user. Some system usernames are restricted (e.g 'mariadb', 'admin', 'standby').",
+							Required:         true,
+							ValidateDiagFunc: validation.ToDiagFunc(validation.All(validation.StringIsNotWhiteSpace, validation.StringLenBetween(1, 63))),
+						},
+						"password": {
+							Type:             schema.TypeString,
+							Description:      "The password for a MariaDB user.",
+							Required:         true,
+							Sensitive:        true,
+							ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotWhiteSpace),
+						},
+					},
+				},
+			},
+			"maintenance_window": {
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Description: "A weekly 4 hour-long window, during which maintenance might occur.",
+				Optional:    true,
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						// TODO -- check if we need to add extra validation here for time
+						"time": {
+							Type:             schema.TypeString,
+							Description:      "Start of the maintenance window in UTC time.",
+							Required:         true,
+							ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotWhiteSpace),
+						},
+						// TODO -- check if providing 'sunday' will lead to an error
+						"day_of_the_week": {
+							Type:             schema.TypeString,
+							Description:      "The name of the week day.",
+							Required:         true,
+							ValidateDiagFunc: validation.ToDiagFunc(validation.IsDayOfTheWeek(true)),
+						},
+					},
+				},
+			},
+			"dns_name": {
+				Type:        schema.TypeString,
+				Description: "The DNS name pointing to your cluster.",
+				Computed:    true,
+			},
+		},
+		Timeouts: &resourceDefaultTimeouts,
+	}
+}
+
+func mariaDBClusterCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(services.SdkBundle).MariaDBClient
+
+	cluster, err := dbaasService.GetMariaDBClusterDataCreate(d)
```

Comment:
naming seems a bit off: both 'Get' and 'Create' 
also, we are using both a `dbaasService` and a `MariaDBClient` objects to perform db functionality so maybe we should integrate these into the same service

Reply:
@cbeti-ionos I agree with this and also with the comments that you provided related to the structure of the packages. I will leave this as it is for now and as soon as this branch is merged, I will start working on separating the dbaas services into different packages and make all the refactor/renaming work. I decided this because splitting dbaas services into different packages it's not the main purpose of this PR.